### ISSUE TITLE
Gracefull ShutDown Not working with mesh

### ISSUE THREAD

**Issue by Pulankit-so** [senecajs/seneca#623]
* Gracefull shutdown behavior is not working when tested with two instances connected over seneca-mesh.Consider two services service1 and service2 . 
*  _seneca (v3.3),seneca-mesh(v0.10.0),seneca-balance-client(v0.6.0)_
**Service1** (with base node) is as follows:-
    ``` javascript
        var seneca = require('seneca')({log:'silent'})
        seneca.add('role:service1,cmd:status',(msg,done)=>{
	      setTimeout(function(){
	         	 done({status:'running'})
	      },500)
        })
        seneca.add('role:service1,cmd:error',(msg,done)=>{
	       throw new Error('I am goint to break things')
        })
        seneca.use('mesh',{
              isbase:true,
              listen:[{pin:'role:service1'}],
              port:39999,
              bases:['127.0.0.1:39999'],
              host:'127.0.0.1'
         })
        seneca.ready(()=>{
	      console.log('service 1 seneca is ready')
	      setTimeout(function(){	
	         	seneca.act({role:'service2',cmd:'status'},(err,data)=>{
	           		console.log('status got called',err,data)
	        	})
	        	seneca.act({role:'service2',cmd:'error'},(err,data)=>{
	         		console.log('error was thrown')
	         	})
	      },10000)
        })
   ```
*  **Service2** is as follows:-
  ``` javascript
        var seneca = require('seneca')({log:'silent'})
        seneca.add('role:service2,cmd:status',(msg,done)=>{
	      setTimeout(function(){
	         	 done({status:'running'})
	      },500)
        })
        seneca.add('role:service2,cmd:error',(msg,done)=>{
	       throw new Error('I am goint to break things')
        })
        seneca.use('mesh',{
              listen:[{pin:'role:service2'}],
              auto: true,
              bases:['127.0.0.1:39999'],
              host:'127.0.0.1'
         })
        seneca.ready(()=>{
	      console.log('service 2 seneca is ready')
        })
   ```

* The output of the act called in service1 should be `status : running` for the first call and timeout error for second. But as soon as the second act is called the process for service2 crashes immediately without returning  result for the first act, hence timeout error occurs for first act as well. Do let me know if this is expected behavior or not.

**Comment by rjrodger** [comment]
It's a bug - services should not crash in this way. Thanks for reporting!

**Comment by rjrodger** [comment]
bug in seneca-transport - will be fixed in next release of seneca-transport

**Comment by Pulankit-so** [comment]
ok thanks :+1:

**Comment by rjrodger** [comment]
@Pulankit-so Thank you again for your great example code - you've inspired my to create a special repo to record and run sample code for bugs: https://github.com/senecajs/repro
This issue is the first entry, and I can confirmed it is fixed in seneca master, so we will have a fix in 3.4.0

**Comment by Pulankit-so** [comment]
Thanks @rjrodger  .Happy to contribute in anyway :-)
